bundler.d/puma.rb

```diff
@@ -0,0 +1,11 @@
+group :puma do
+  if RUBY_VERSION < '2.3'
+    # Important!
+    # The last actual version that supports v2.0.0 up to v2.2.0 is 3.10.0
+    # Puma version 3.11.0 changed the usage of socket to a feature that is not
+    # supported by Ruby 2.2.0 and lower, and it causes a crash on TLS!
+    gem 'puma', '~>3.10', :require => 'puma'
```

Comment:
`~>3.10` means `> 3.10 and < 4.0`, you probably want `~>3.10.0`

Reply:
Ok should I keep ~>3.12 then?